### ISSUE TITLE
incorrect error on inline template

### ISSUE THREAD

**Issue by walkamongus** [rodjek/puppet-lint#545]
a `[single_quote_string_with_variables] single quoted string containing a variable found` error is thrown on the following use of an inline template:

```
$list = ['one', 'two', 'three']
file { '/tmp/text.txt':
  ensure  => file,
  content => inline_epp('<% $list.each |$item| { %><%= "${item}\n" %><% } %>'),
}
```

Should puppet-lint ignore the contents of an inline_epp/inline_template call?

**Comment by rnelson0** [comment]
Not only that, it should ignore the contents of single quoted strings which have no interpolation.

**Comment by rnelson0** [comment]
To expand on that: You COULD have it try and evaluate it, but it won't have the context to know if it's part of the variables in the interpreted part of the template, or a string literal, and it's probably best not to try and make assumptions that may not hold valid.
